#162199433 Create featured-article routes

TheoOkafor opened the pull request:
#### What does this PR do?
Create select-hero-article route

#### Description of Task to be completed?
- Create route for admin to select the featured article
- Create logic for auto-selection of featured article for admin
- Create route for fetching the featured article.
- Write tests for the routes
- Document the routes

#### How should this be manually tested?
- git clone `https://github.com/andela/haven-ah-backend.git`.
- Install `node` if not installed.
- ensure that PostgreSQL is running on your local machine.
- run `npm install` to install the npm packages.
- run `npm test` to run the unit and integration tests.

To test on Postman
- run `npm run start-dev`
- Add the title, body and description as `x-www-form-urlencoded`
- On postman run:
 PUT
 `localhost:5000/api/v1/admin/articles/featured` (set JWT token as`x-access-token` in header) to select a featured article
GET
`localhost:5000/api/v1/articles/featured` to fetch a featured article

NB: You need admin rights to access the PUT route.

#### Any background context you want to provide?
This route is required for selecting the article that will be displayed on the hero-image of the landing page.
The article can be auto-selected and it can be selected by the admin. Either way, the admin has to initiate the selection through this route.

#### What are the relevant pivotal tracker stories?
#162199433

#### Screenshots (if appropriate)

Admin-select:
<img width="1046" alt="screen shot 2018-11-27 at 1 00 10 pm" src="https://user-images.githubusercontent.com/31534129/49080591-7508a980-f244-11e8-81a2-200df80334c8.png">


Auto-select:
<img width="1031" alt="screen shot 2018-11-27 at 1 00 21 pm" src="https://user-images.githubusercontent.com/31534129/49080597-7b972100-f244-11e8-9e28-e7febe2c9fee.png">

TheoOkafor committed:
feat(article): Create select hero article route

- Create route for admin to select the hero article
- Create logic for auto-selection of hero article for admin
- Write test for the route
- Write documentation for the route

[#162199433]

victor-nweze-deel commented on server/controllers/article.js:
```diff
@@ -219,6 +220,70 @@ class Article {
+        topArticles.push(allArticles[i].dataValues);
+      }
+    }
+    const randomIndex = Math.floor(Math.random() * topArticles.length);
```
nice

TheoOkafor replied:
Thanks

victor-nweze-deel reviewed (commented):
Good Job @TheoOkafor

ucheg6 commented:
Good job @TheoOkafor. 👍

TheoOkafor commented:
> Good job @TheoOkafor. 👍

Thank you @ucheg6

TheoOkafor committed:
feat(article): Create get featured article route

- Create route for fetching the featured article
- Write test for the route
- Write documentation for the route
- Add CORS to the app

[#162199433]
